amundsen_application/static/js/components/common/EditableSection/index.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { OverlayTrigger, Popover } from 'react-bootstrap';
+
+import './styles.scss';
+
+export interface EditableSectionProps {
+  title: string;
+  readOnly?: boolean;
+  /* Should be used when readOnly=true to prompt users with a relevant explanation for the given use case */
+  editText?: string;
+  /* Should be used when readOnly=true to link to the source where users can edit the given metadata */
+  editUrl?: string;
+}
+
+interface EditableSectionState {
+  isEditing: boolean;
+}
+
+export interface EditableSectionChildProps {
+  isEditing?: boolean;
+  setEditMode?: (isEditing: boolean) => void;
+}
+
+export class EditableSection extends React.Component<EditableSectionProps, EditableSectionState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isEditing: false,
+    }
+  }
```

Comment:
Nit: It depends a bit on the babel configuration of the project, but I usually found more readable and easy to set the state as an instance variable outside the constructor, like this:
```
export class EditableSection extends ... {
  state = {isEditing: false};
...
```
This also avoids the common mistake of forgetting to call 'super'!

Reply:
Thanks for the comment, I hadn't seen this yet. I'll read about it more to see if there's other benefits.